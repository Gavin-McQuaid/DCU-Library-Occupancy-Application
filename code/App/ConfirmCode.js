import React, {Component} from 'react';
import {View,Text, Alert, TextInput, StyleSheet, Image,KeyboardAvoidingView, TouchableOpacity} from 'react-native';
import {hook} from 'cavy';
class ConfirmCode extends Component {
	static navigationOptions = {
		title : 'ConfirmCode'
	}

	constructor(props){
		super(props)
		this.state = {
			ResetCode: ''
		}
	}
    
    // Function that confirms an inputted password reset code with the server.
	ConfirmCode = () => {
		const {ResetCode} = this.state;
		fetch('http://ec2-34-244-240-250.eu-west-1.compute.amazonaws.com/ConfirmCode.php', {
			method: 'POST',
			headers: {
				'Accept' : 'application/json',
				'Content-Type' : 'applcation/json'
			},
			body: JSON.stringify({
				code: ResetCode,
				email: this.props.navigation.state.params.email
			}) 
		}).then((response) => response.json())
      .then((responseJson) => {

		if(responseJson == "Success"){
			this.props.navigation.navigate("ResetPassword", {screen: "ResetPassword", email: this.props.navigation.state.params.email});
		}
		else{
			Alert.alert("That code is incorrect or expired");
		}

      }).catch((error) => {
        console.error(error);
      });

	}


	
	render() {
		return (
		<KeyboardAvoidingView behavior = 'padding' style = {styles.container}>
			<View style = {styles.container}>
				<View style = {styles.logoContainer}>
					<Text style = {styles.title}> DCU Library App 
					</Text>
					<Image
						style = {styles.logo}
						source = {require('./images/logo.png')} 
					/>				
				</View>
				<TextInput
					ref = {this.props.generateTestHook('ConfirmCode.TextInput')}
					placeholder = "6 Digit Number"
					placeholderTextColor = 'white'
					returnKeyType = 'next'
					autoCapitalize = 'none'
					autoCorrect = {false}
					onChangeText ={ResetCode => this.setState({ResetCode})}
					style = {styles.input}
				/>		
				<TouchableOpacity
					ref = {this.props.generateTestHook('ConfirmCode.Button')}
					style = {styles.buttonContainer}
					onPress={() => this.ConfirmCode()}>
					<Text style = {styles.buttontext}> Confirm code 
					</Text>
				</TouchableOpacity>
			</View>
		</KeyboardAvoidingView>	
		);
	}
}

export default hook(ConfirmCode);
const styles = StyleSheet.create({
  container: {
	backgroundColor : 'cornflowerblue',  
    flex: 1,

  },

  logoContainer: {
	alignItems : 'center',
	flexGrow : 1,
	justifyContent : 'center'
  },

  logo: {
	width : 200,
	height : 150
  },

  title: {
	color : 'white',
	marginBottom : 10,
	fontSize : 30,
	opacity : 0.9,
	justifyContent: 'center'
	
  },

  input: { 
	height: 40,
	backgroundColor : 'rgba(255,255,255,0.2)',
	marginBottom : 15,
	color : 'white',
	paddingHorizontal : 10,
	borderRadius : 25
	},

  buttonContainer: {
	backgroundColor :'steelblue',
	paddingVertical : 15,
	marginBottom : 15,
	borderRadius : 25
	},

  buttontext: {
	color : 'white',
	textAlign : 'center',
	fontWeight : '400'
	}
});