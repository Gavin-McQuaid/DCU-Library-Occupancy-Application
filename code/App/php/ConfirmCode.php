<?php include '../inc/dbinfo.inc'; 

  /* Connect to MySQL and select the database. */
  $connection = mysqli_connect(DB_SERVER, DB_USERNAME, DB_PASSWORD);

  if (mysqli_connect_errno()) echo 'Failed to connect to MySQL: ' . mysqli_connect_error();

  $database = mysqli_select_db($connection, DB_DATABASE);
  $json = file_get_contents('php://input');
  $jsonobject = json_decode($json,true);
  
  $email = $jsonobject['email'];
  $code = intval($jsonobject['code']);
  $time = time(); 
  $query = "SELECT * FROM password_reset_requests  WHERE email = '$email' AND reset_code = '$code' AND ('$time' - timestamp) <= 300";
  $doCheck = mysqli_fetch_array(mysqli_query($connection,$query));
  if(isset($doCheck)){
	$resetSuccessMsg = 'Success';
	$jsonEncodeMsg = json_encode($resetSuccessMsg);
	echo $jsonEncodeMsg;

}
else{
$resetFailureMsg = 'That code is incorrect or expired';
$jsonEncodeFailure = json_encode($resetFailureMsg);
echo $jsonEncodeFailure;
}

mysqli_close($connection);
?>
