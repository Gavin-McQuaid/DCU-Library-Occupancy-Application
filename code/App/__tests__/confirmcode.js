import React from 'react';
import ConfirmCode from '../ConfirmCode';
import renderer from 'react-test-renderer';
import {shallow, mount,render,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockStore from 'redux-mock-store';
const Text = require('Text');
const TouchableHighlight = require('TouchableHighlight');
const ReactTestRenderer = require('react-test-renderer');
const TouchableOpacity = require('TouchableOpacity');
const TextInput = require('TextInput');
const Image = require('Image');

var ReactTestUtils = require('react-dom/test-utils')
configure({ adapter: new Adapter() })

describe('Confirm Code', () =>{
	it('text shows', () =>{
		const instance0 = ReactTestRenderer.create(
			<Text style = {{}}> DCU Library App </Text>	
		);
		});
	it('image shows', () =>{
		const instance1 = ReactTestRenderer.create(
			<Image style = {{}} source = {require('./Images/logo.png')}> </Image>
		);
	});
	it('textinput shows', () =>{
		const instance2 = ReactTestRenderer.create(
			<TextInput placeholder = '6 Digit Number'> </TextInput>
		);
	});
	it('touchable opacity shows', () =>{
		const instance3 = ReactTestRenderer.create(
			<TouchableOpacity style ={{}}>
			<Text> Create Account </Text>
			</TouchableOpacity>
		);
	});
	it ('Text Input	works', () => {
		const wrapper = shallow(<ConfirmCode />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const credentials = {code : '123456'};
		TextInput1.value = credentials.code;
		expect(TextInput1.value).toBe('123456');
	});
	it('touchableOpacity 1 touch works', () => {
		const wrapper = shallow(<ConfirmCode />);
		const touchOpacity1 = wrapper.find('TouchableOpacity').at(0);
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<ConfirmCode store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
	it ('Confirm Code works', () => {
		const wrapper = shallow(<ConfirmCode />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const credentials = {code : '123456'};
		TextInput1.value = credentials.code;
		expect(TextInput1.value).toBe('123456');
		const touchOpacity1 = wrapper.find('TouchableOpacity').at(0);
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<ConfirmCode store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
		const rendered = renderer.create(<ConfirmCode/>).toJSON();
		expect(rendered).toMatchSnapshot();
		
	
});