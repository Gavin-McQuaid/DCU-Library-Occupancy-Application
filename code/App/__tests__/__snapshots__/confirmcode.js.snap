// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="6 Digit Number"
      placeholderTextColor="white"
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Confirm code
      </Text>
    </View>
  </View>
</View>
`;

exports[`Confirm Code Confirm Code works 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="6 Digit Number"
      placeholderTextColor="white"
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Confirm code
      </Text>
    </View>
  </View>
</View>
`;

exports[`Confirm Code touchableOpacity 1 touch works 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="6 Digit Number"
      placeholderTextColor="white"
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Confirm code
      </Text>
    </View>
  </View>
</View>
`;
